rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -785,137 +815,209 @@ def test_missing_acquisitions(accountant):
+    add_acquisition(pot, ONE, price=FVal(200))
+    expected_owned_asset_value += events[1].taxable_amount * events[1].price
+    expected_remaining_amount += events[1].taxable_amount
+    assert_properties()
+    add_spend(pot, FVal(2))  # Spend the entire amount acquisitions
+    last_cost_basis = events[-1].cost_basis
+    assert last_cost_basis.taxable_bought_cost == expected_owned_asset_value  # The entire amount
+    assert last_cost_basis.is_complete is True
+    expected_owned_asset_value = ZERO
+    expected_remaining_amount = ZERO
+    assert_properties()
+
+
+def test_accounting_average_cost_basis_missing_acquisitions(accountant):
+    """
+    See that using more than the available acquisitions in average cost basis
+    adds a MissingAcquisition.
+    """
+    pot = accountant.pots[0]
+    events = pot.processed_events
+    cost_basis = pot.cost_basis
+    cost_basis.reset(DBSettings(cost_basis_method=CostBasisMethod.ACB))
+    asset_events = cost_basis.get_events(A_ETH)
+    manager = asset_events.acquisitions_manager
+    expected_owned_asset_value = ZERO
+    expected_remaining_amount = ZERO
+
+    def assert_properties():
+        assert manager.current_owned_asset_value == expected_owned_asset_value
+        assert manager.remaining_amount == expected_remaining_amount
+
+    add_acquisition(pot, FVal(2))
+    expected_owned_asset_value = events[0].taxable_amount * events[0].price
+    expected_remaining_amount = events[0].taxable_amount
+    assert_properties()
+    add_spend(pot, FVal(3.5))
+    last_cost_basis = events[-1].cost_basis
+    assert last_cost_basis.is_complete is False
+    assert last_cost_basis.taxable_bought_cost == expected_owned_asset_value  # The entire amount
     assert cost_basis.missing_acquisitions == [
         MissingAcquisition(
-            asset=asset,
-            time=0,
-            found_amount=FVal(2),
-            missing_amount=FVal(1.5),
+            asset=A_ETH,
+            time=Timestamp(1675483017),
+            found_amount=FVal(expected_remaining_amount),
+            missing_amount=FVal(events[-1].taxable_amount - expected_remaining_amount),
         ),
     ]
+    expected_owned_asset_value = ZERO
+    expected_remaining_amount = ZERO
+    assert_properties()
+
+
+def test_accounting_average_cost_basis_not_taxable(accountant):
+    """
+    Tests that when there are non-taxable events and `reduce_asset_amount` is used, cost basis
+    doesn't break.
+    """
+    pot = accountant.pots[0]
+    events = pot.processed_events
+    cost_basis = pot.cost_basis
+    cost_basis.reset(DBSettings(cost_basis_method=CostBasisMethod.ACB))
+    asset_events = cost_basis.get_events(A_ETH)
+    manager = asset_events.acquisitions_manager
+    expected_owned_asset_value = ZERO
+    expected_remaining_amount = ZERO
+
+    def assert_properties():
+        assert manager.current_owned_asset_value == expected_owned_asset_value
+        assert manager.remaining_amount == expected_remaining_amount
```

Comment:
this code repeats around all the tests in this file. Maybe you can make it a normal function and take as arguments the manager and the two amounts and just pass the arguments. This avoid repetition and possible mistakes

Reply:
hmm, maybe can be changed to a normal function, yes